File: Two Pointers/Java/Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder str=new StringBuilder();
        for(char i:s.toCharArray()){
            if(Character.isLetterOrDigit(i))str.append(Character.toLowerCase(i));         
        }
        StringBuilder cmp=new StringBuilder(str);
        cmp.reverse();
        return str.toString().equals(cmp.toString());
    }
}

File: Two Pointers/Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str;
        for(auto i:s){
            if(!isalnum(i))continue;
            else if(isupper(i))str+=tolower(i);
            else str+=i;
        }
        string cmp=str;
        reverse(str.begin(),str.end());
        return cmp==str;
    }
};
